workspaces/topology/plugins/topology/src/utils/download-log-file-utils.ts

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * Triggers a download of the given string data as a file with the specified filename.
+ *
+ * @param data - The content to download
+ * @param filename - The name of the file to save as
+ */
+export const downloadLogFile = (data: string, filename: string): void => {
+  const blob = new Blob([data], { type: 'text/plain;charset=utf-8' });
+  const url = URL.createObjectURL(blob);
+  const link = document.createElement('a');
+  link.href = url;
+  link.download = filename;
+
+  document.body.appendChild(link); // Required for Firefox
```

Comment:
we can probably simply this function a bit now because this is no longer required for new versions of firefox. this is what I used for the PatternFly `CodeEditor` component:

```js
/**
 * Downloads a file to a users device given its string content and a full file name.
 */
const defaultOnDownload = (value: string, fileName: string) => {
  const link = document.createElement('a');
  link.href = URL.createObjectURL(new Blob([value], { type: 'text' }));
  link.download = fileName;
  link.click();
};
```